File: omp/examples/scheduling/schedule.c
#include <stdio.h>
#include <unistd.h>
#include <omp.h>

int main(){

  #pragma omp parallel for
  for (int i=0; i < 10; i++){
    printf("handling number %d from thread %d\n", i, omp_get_thread_num());
  }

  sleep(1);
  printf("next with different schedule policy. decided at compile time\n");
  sleep(1);

  #pragma omp parallel for schedule(static, 2)
  for (int i=0; i < 10; i++){
    printf("handling number %d from thread %d\n", i, omp_get_thread_num());
  }

}

File: omp/examples/scheduling/schedule.f90
PROGRAM schedule
   USE omp_lib
   IMPLICIT NONE

   INTEGER :: i

   !$OMP PARALLEL DO
   DO i = 0, 9
      WRITE(*,'(A,I0,A,I0)') 'handling number ', i, ' from thread ', omp_get_thread_num()
   END DO
   !$OMP END PARALLEL DO

   CALL sleep(1)
   WRITE(*,*) 'next with different schedule policy. decided at compile time'
   CALL sleep(1)

   !$OMP PARALLEL DO SCHEDULE(STATIC, 2)
   DO i = 0, 9
      WRITE(*,'(A,I0,A,I0)') 'handling number ', i, ' from thread ', omp_get_thread_num()
   END DO
   !$OMP END PARALLEL DO

END PROGRAM schedule